cw_solana/lib/solana_client.dart

```diff
@@ -0,0 +1,490 @@
+      } else {
+        webSocketUrl = 'wss://${node.uriRaw}';
+      }
+
+      _client = SolanaClient(
+        rpcUrl: isModifiedNodeUri ? rpcUri! : node.uri,
+        websocketUrl: Uri.parse(webSocketUrl),
+        timeout: const Duration(minutes: 2),
+      );
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  Future<double> getBalance(String address) async {
+    try {
+      final balance = await _client!.rpcClient.getBalance(address);
+
+      final solBalance = balance.value / lamportsPerSol;
+
+      return solBalance;
+    } catch (_) {
+      return 0.0;
+    }
+  }
+
+  Future<ProgramAccountsResult?> getSPLTokenAccounts(String mintAddress, String publicKey) async {
+    try {
+      final tokenAccounts = await _client!.rpcClient.getTokenAccountsByOwner(
+        publicKey,
+        TokenAccountsFilter.byMint(mintAddress),
+        commitment: Commitment.confirmed,
+        encoding: Encoding.jsonParsed,
+      );
+      return tokenAccounts;
+    } catch (e) {
+      return null;
+    }
+  }
+
+  Future<SolanaBalance?> getSplTokenBalance(String mintAddress, String publicKey) async {
+    // Fetch the token accounts (a token can have multiple accounts for various uses)
+    final tokenAccounts = await getSPLTokenAccounts(mintAddress, publicKey);
+
+    // Handle scenario where there is no token account
+    if (tokenAccounts == null || tokenAccounts.value.isEmpty) {
+      return null;
+    }
+
+    // Sum the balances of all accounts with the specified mint address
+    double totalBalance = 0.0;
+
+    for (var programAccount in tokenAccounts.value) {
+      final tokenAmountResult =
+          await _client!.rpcClient.getTokenAccountBalance(programAccount.pubkey);
+
+      final balance = tokenAmountResult.value.uiAmountString;
+
+      final balanceAsDouble = double.tryParse(balance ?? '0.0') ?? 0.0;
+
+      totalBalance += balanceAsDouble;
+    }
+
+    return SolanaBalance(totalBalance);
+  }
+
+  Future<double> getGasForMessage(String message) async {
+    try {
+      final gasPrice = await _client!.rpcClient.getFeeForMessage(message) ?? 0;
+      final fee = gasPrice / lamportsPerSol;
+      return fee;
+    } catch (_) {
+      return 0;
+    }
+  }
+
+  /// Load the Address's transactions into the account
+  Future<List<SolanaTransactionModel>> fetchTransactions(
+    Ed25519HDPublicKey publicKey, {
+    String? splTokenSymbol,
+    int? splTokenDecimal,
+  }) async {
+    List<SolanaTransactionModel> transactions = [];
+
+    // Using mainnet beta node to fetch transactions.
+    final mainNetBetaConfig = {"uri": "api.mainnet-beta.solana.com"};
+
+    final mainNetBetaNode = Node.fromMap(mainNetBetaConfig);
+    mainNetBetaNode.type = WalletType.solana;
+
+    final mainNetBetaSolanaClient = SolanaClient(
+      rpcUrl: mainNetBetaNode.uri,
+      websocketUrl: Uri.parse('wss://${mainNetBetaNode.uriRaw}'),
+      timeout: const Duration(minutes: 2),
+    );
```

Comment:
why are we still using the mainnet-beta node for this? is the ankr node not fetching the balances correctly?

Reply:
Ankr node works perfectly

I've switched to fully depend on it